### ISSUE TITLE
parallel or SPI driven

### ISSUE THREAD

**Issue by df8oe** [sp9bsl/UHSDR#6]
Hi Slawek,

I think I will have my parallel 3.5" LCD ready for the weekend. It will be a "hack" but functional identical to final soultion:

I will desolder all CMOS logic ICs which are on green PCB. Only XPT2046 and peripheral will stay. CS and RS lines are already present on RPI connector. D0...D15 are at outputs of the two shift register IC. I will check if RD / WR is connected to logic gates (14pin IC). I think it will be maximum 20 wires which I must fit from PCB to my adaptor PCB I already do have. So I will have 3.5" LCD parallel which has connectors for mcHF AND OVI40 - very good for testing. Question: Do you think you can activate parallel drive instead of SPI "blind" - and I try it? I do not want to spend much time in SPI on OVI40 because it will be parallel in final...

**Comment by sp9bsl** [comment]
I'll try my best to help you. I can ofcourse do the same connection to lcd but test it on mchf.

**Comment by df8oe** [comment]
If you do have time: of course this would help. If it works on mcHF the step to port it to OVI40 is small.

**Comment by sp9bsl** [comment]
Time is a "relative" term :) Of course it is limited, but before i build adapter it is good idea to test connections this way.

**Comment by s53dz** [comment]
Hi Andreas,

Just checking that I understand you correctly:
you go for a parallel LCD interface in OVI40 to:
- gain speed and
- lower the (helicopter) interference frequency and level?

TNX 73 Bojan

**Comment by df8oe** [comment]
YES. Absolutely correct. OVI40 standard LCD mode was and is PARALLEL.

73
Andreas

**Comment by s53dz** [comment]
OK, I see. So you did it for mcHF only because of the lack of STM32F4 GPIO pins, I assume.

73 Bojan

**Comment by sp9bsl** [comment]
Hi Andreas,
the activation of parallel interface isn't so easy. It requires completelly rewrite of whole init function. I will start at saturday evening. There is RD pin on lcd pcb but connected to VCC, it would be good to connect it to UI RD but it requires disassembly of LCD. I would propose to not to do that. I'll try "blind" transmit. so we simply use #ifdef. I almost finished adapter board but need some tests (wired connection to parallel). Let me know if you are ready.
73 Slawek

**Comment by df8oe** [comment]
Hi Slawek,
I know that there are some (!!) tracks that must be cut... I have cut Vcc to backlight LED already at my SPI adaptored LCD so that I can dim backlight. That's a point I forgot to tell you: we need the possibility of dimming backlight via PWM. There is more than possibility to get that. Easiest is to  add a small switching MOSFET between cathodes of LEDs and GND and drive its gate with PWM. II will cut RD, too using the same procedure. It is not so difficult: I cut it using a very sharp knife at the point of soldering and rewired the remaining soldering on the flat ribbon using thin wire. Unfortunately my new LCD which I I need for building new adaptor has not reached yet - I hope I will get it today. Of course I write you if I have got it.

SPI does work very well on OVI40, too except the slow speed due to missing SPI_DMA. But, again: I do not see that we invest much time and effort to get it work because of OVI40 LCD will run in parallel mode.

73
Andreas

**Comment by sp9bsl** [comment]
Ok, i understand we'll go into parallel. I want to add autodetection of interface SPI/parallel (and ofcourse repair SPI DMA for F7) but not now, i promised some people that SPI will work either. I will send you a schematic file of new adapter in a while, i used there PWM dimming already. About RD - OK i'll do the same and provide wire to adapter to test it.

73, Slawek

**Comment by df8oe** [comment]
Supporting both SPI and parallel is welcome of course, too. We do not know for what we can/will use it in future and code size is absoulutely no problem at OVI40. I am thinking about autodetection of different (!!) LCD types on OVI40 and select the correct driver at runtime. I think we should do that on OVI40 - Danilo, are you reading here?

73
Andreas

**Comment by sp9bsl** [comment]
Hi Andreas,
I think that my repo is now ok (of course thanks to Danilo!), now i have v2.5.127.
I modified code of ILI9486 init function to support parallel interface. It works with spi and should with parallel too, please simply disable the USE_DISPLAY_SPI. Unfortunatelly i didn't check this in real because i still haven't lcd connected with parallel interface.

73
Slawek

**Comment by sp9bsl** [comment]
Hi Andreas,
Autodetection works, command 0xD3 returned 0x9486 from my lcd in mcHF. You may try the latest code, it should work for mcHF if you disconnect SPI CLK from WR pin of lcd (partially remove the UI-04-N-025), and i hope for OVI40 it will work too.

73 Slawek

**Comment by df8oe** [comment]
Very good. I will test tomorrow with OVI40. BTW it is quite normal to remove short circuit of parallel and SPI when parallel is used. That is neccessary for standard mcHF LCD, too.

73
Andreas
